arbos/constraints/model.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+// The constraints package tracks the multi-dimensional gas usage to apply constraint-based pricing.
+package constraints
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/offchainlabs/nitro/util/arbmath"
+)
+
+const (
+	PricingInertiaFactor = 30
+)
+
+type PricingState struct {
+	constraints ResourceConstraints
+}
+
+// UpdatePricingModel adjusts the basefee according to simplified constraint-based pricing.
+// Formula: basefee = F_min * exp( max_i ( B_i / (30 * T_i * sqrt(Δ_i)) ) )
+func (ps *PricingState) UpdatePricingModel(minBaseFee *big.Int, timePassed uint64) *big.Int {
+	var maxExponentBips arbmath.Bips
+
+	for c := range ps.constraints.All() {
+		// Decay per-constraint backlog by T_i * timePassed
+		c.RemoveFromBacklog(timePassed)
+
+		if c.Backlog == 0 || c.TargetPerSec == 0 || c.Period == 0 {
+			continue
+		}
+
+		// Compute inertia = 30 * sqrt(Δ_i)
+		inertia := PricingInertiaFactor * uint64(math.Floor(math.Sqrt(float64(c.Period))))
```

Comment:
I'm not sure whether we can have floating-point math in Arbos. However, if it is fine, we could convert from float back to int only at the end of the calculation, thereby minimizing error propagation in the formula.